sql/expression/function/json/json_keys_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/src-d/go-errors.v1"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+func TestJSONKeys(t *testing.T) {
+	_, err := NewJSONKeys()
+	require.True(t, errors.Is(err, sql.ErrInvalidArgumentNumber))
+
+	f1 := buildGetFieldExpressions(t, NewJSONKeys, 1)
+	f2 := buildGetFieldExpressions(t, NewJSONKeys, 2)
+
+	testCases := []struct {
+		f   sql.Expression
+		row sql.Row
+		exp interface{}
+		err bool
+	}{
+		{
+			f:   f1,
+			row: sql.Row{`null`},
+			exp: nil,
+		},
+		{
+			f:   f1,
+			row: sql.Row{`1`},
+			exp: nil,
+		},
+		{
+			f:   f1,
+			row: sql.Row{`[1]`},
+			exp: nil,
+		},
+		{
+			f:   f1,
+			row: sql.Row{`{}`},
+			exp: types.MustJSON(`[]`),
+		},
+		{
+			f:   f1,
+			row: sql.Row{`"fjsadflkd"`},
+			err: true,
```

Comment:
I noticed some differences in what MySQL considers valid JSON that seem to disagree with this test case...

When passing MySQL the string `fjsadflkd`, I get an error, but when passing a string with that value quoted (`"fjsadflkd"`) I get `NULL`:
```
mysql> select JSON_KEYS("fjsadflkd");
ERROR 3141 (22032): Invalid JSON text in argument 1 to function json_keys: "Invalid value." at position 1.

mysql> select JSON_KEYS('"fjsadflkd"');
+--------------------------+
| JSON_KEYS('"fjsadflkd"') |
+--------------------------+
| NULL                     |
+--------------------------+
1 row in set (0.00 sec)
```

Reply:
I didn't see the double quote inside of the test input, and is why I added the `case string:` statement here:
https://github.com/dolthub/go-mysql-server/pull/2304#discussion_r1473393487

I've removed it now, and added tests for inputs a string and a string containing a quotes.